src/scripts/main.js

```diff
@@ -1,3 +1,273 @@
+optionSingapore.textContent = 'Singapore';
+officeInput.appendChild(optionSingapore);
+
+const optionLondon = document.createElement('option');
+
+optionLondon.value = 'London';
+optionLondon.textContent = 'London';
+officeInput.appendChild(optionLondon);
+
+const optionNewYork = document.createElement('option');
+
+optionNewYork.value = 'New York';
+optionNewYork.textContent = 'New York';
+officeInput.appendChild(optionNewYork);
+
+const optionEdinburgh = document.createElement('option');
+
+optionEdinburgh.value = 'Edinburgh';
+optionEdinburgh.textContent = 'Edinburgh';
+officeInput.appendChild(optionEdinburgh);
+
+const optionSanFrancisco = document.createElement('option');
+
+optionSanFrancisco.value = 'San Francisco';
+optionSanFrancisco.textContent = 'San Francisco';
+officeInput.appendChild(optionSanFrancisco);
+
+const age = document.createElement('label');
+
+age.textContent = 'age: ';
+
+const ageInput = document.createElement('input');
+
+ageInput.type = 'text';
+ageInput.setAttribute('data-qa', 'age');
+ageInput.required = true;
+
+const salary = document.createElement('label');
+
+salary.textContent = 'salary: ';
+
+const salaryInput = document.createElement('input');
+
+salaryInput.type = 'text';
+salaryInput.setAttribute('data-qa', 'salary');
+salaryInput.required = true;
+
+form.classList.add('new-employee-form');
+
+document.body.appendChild(form);
+
+form.appendChild(namelabel);
+namelabel.appendChild(nameInput);
+
+form.appendChild(position);
+position.appendChild(positionInput);
+
+form.appendChild(office);
+office.appendChild(officeInput);
+
+form.appendChild(age);
+age.appendChild(ageInput);
+
+form.appendChild(salary);
+salary.appendChild(salaryInput);
+
+const button = document.createElement('button');
+
+button.textContent = 'Save to table';
+button.type = 'submit';
+form.appendChild(button);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const tbody = document.querySelector('tbody');
+
+  const newRow = document.createElement('tr');
+
+  const newName = document.createElement('td');
+  const newPosition = document.createElement('td');
+  const newOffice = document.createElement('td');
+  const newAge = document.createElement('td');
+  const newSalary = document.createElement('td');
+
+  if (nameInput.value.length < 4) {
+    pushNotification(
+      10,
+      10,
+      'Error',
+      'Error\n ' + 'name can`t be less than 4 characters',
+      'error',
+    );
+
+    return;
+  }
+  newName.textContent = nameInput.value;
+  newPosition.textContent = positionInput.value;
+  newOffice.textContent = officeInput.value;
+
+  if (!isNaN(ageInput.value) && ageInput.value.trim() !== '' && +ageInput.value > 18 && +ageInput.value < 90) {
+    newAge.textContent = ageInput.value;
+  } else {
+    pushNotification(
+      10,
+      10,
+      'Error',
+      'Error\n ' + 'age must be number and more than 18 and less than 90',
+      'error',
+    );
+
+    return;
+  }
+
```

Comment:
The condition mistakenly checks 'ageInput.value' instead of 'salaryInput.value'. This should be corrected to properly validate the salary input.